code/game/objects/items/gift.dm

```diff
@@ -1,116 +1,123 @@
+	if(HAS_MIND_TRAIT(user, TRAIT_CANNOT_OPEN_PRESENTS))
+		to_chat(user, span_warning("You're supposed to be spreading gifts, not opening them yourself!"))
 		return
 
-	qdel(src)
+	moveToNullspace()
+
+	var/obj/item/thing = new contains_type(get_turf(user))
 
-	var/obj/item/I = new contains_type(get_turf(M))
-	if (!QDELETED(I)) //might contain something like metal rods that might merge with a stack on the ground
-		M.visible_message(span_notice("[M] unwraps \the [src], finding \a [I] inside!"))
-		M.investigate_log("has unwrapped a present containing [I.type].", INVESTIGATE_PRESENTS)
-		M.put_in_hands(I)
-		I.add_fingerprint(M)
+	if (QDELETED(thing)) //might contain something like metal rods that might merge with a stack on the ground
+		user.visible_message(span_danger("Oh no! The present that [user] opened had nothing inside it!"))
 	else
-		M.visible_message(span_danger("Oh no! The present that [M] opened had nothing inside it!"))
-
-/obj/item/a_gift/proc/get_gift_type()
-	var/gift_type_list = list(/obj/item/sord,
-		/obj/item/storage/wallet,
-		/obj/item/storage/photo_album,
-		/obj/item/storage/box/snappops,
-		/obj/item/storage/crayons,
-		/obj/item/storage/backpack/holding,
-		/obj/item/storage/belt/champion,
-		/obj/item/soap/deluxe,
-		/obj/item/pickaxe/diamond,
-		/obj/item/pen/invisible,
-		/obj/item/lipstick/random,
-		/obj/item/grenade/smokebomb,
-		/obj/item/grown/corncob,
-		/obj/item/poster/random_contraband,
-		/obj/item/poster/random_official,
-		/obj/item/book/manual/wiki/barman_recipes,
-		/obj/item/book/manual/chef_recipes,
-		/obj/item/bikehorn,
-		/obj/item/toy/beach_ball,
-		/obj/item/toy/basketball,
-		/obj/item/banhammer,
-		/obj/item/food/grown/ambrosia/deus,
-		/obj/item/food/grown/ambrosia/vulgaris,
-		/obj/item/pai_card,
-		/obj/item/instrument/violin,
-		/obj/item/instrument/guitar,
-		/obj/item/storage/belt/utility/full,
-		/obj/item/clothing/neck/tie/horrible,
-		/obj/item/clothing/suit/jacket/leather,
-		/obj/item/clothing/suit/jacket/leather/biker,
-		/obj/item/clothing/suit/costume/poncho,
-		/obj/item/clothing/suit/costume/poncho/green,
-		/obj/item/clothing/suit/costume/poncho/red,
-		/obj/item/clothing/suit/costume/snowman,
-		/obj/item/clothing/head/costume/snowman,
-		/obj/item/stack/sheet/mineral/coal)
-
-	gift_type_list += subtypesof(/obj/item/clothing/head/collectable)
-	gift_type_list += subtypesof(/obj/item/toy) - (((typesof(/obj/item/toy/cards) - /obj/item/toy/cards/deck) + /obj/item/toy/figure + /obj/item/toy/ammo)) //All toys, except for abstract types and syndicate cards.
+		user.visible_message(span_notice("[user] unwraps \the [src], finding \a [thing] inside!"))
+		user.investigate_log("has unwrapped a present containing [thing.type].", INVESTIGATE_PRESENTS)
+		user.put_in_hands(thing)
+		thing.add_fingerprint(user)
 
-	var/gift_type = pick(gift_type_list)
+	qdel(src)
 
-	return gift_type
+/obj/item/gift/proc/get_gift_type()
+	var/static/list/gift_type_list = null
+
+	if(isnull(gift_type_list))
```

Comment:
why do you have an `isnull` check here after you explicitly set it to null?

Reply:
Because its a static list? Its not going to be null after we generate the list after that code executes.

Reply:
If the concept is new to you, look at what the old code was doing with the GLOB and the .len check